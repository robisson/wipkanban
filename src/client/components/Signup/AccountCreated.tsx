import React from "react";
import {
  withStyles,
  Theme,
  StyleRulesCallback
} from "@material-ui/core/styles";
import Grid from "@material-ui/core/Grid";
import Paper from "@material-ui/core/Paper";
import Typography from "@material-ui/core/Typography";
import { Link } from "react-router-dom";
import green from "@material-ui/core/colors/green";

const styles: StyleRulesCallback = (theme: Theme) => ({
  root: {
    flexGrow: 1
  },
  paper: {
    padding: theme.spacing.unit * 2,
    height: "100vh",
    display: "flex",
    alignItems: "center"
  },
  backgroundPrimary: {
    backgroundColor: theme.palette.primary.main
  },
  text: {
    width: "100%"
  },
  link: {
    textDecoration: "none",
    color: theme.palette.secondary.main,
    "&:hover": {
      textDecoration: "underline"
    }
  },
  textSuccessfull: {
    color: green[500]
  }
});

interface IAccountCreatedProps {
  classes: any;
}

const AccountCreated: React.StatelessComponent<any> = ({
  classes
}: IAccountCreatedProps) => {
  return (
    <Grid container className={classes.root}>
      <Grid item xs={12} className={classes.demo}>
        <Grid
          container
          className={classes.demo}
          direction="row"
          justify="flex-start"
        >
          <Grid item xs={12} sm={12} md={7} lg={8} xl={8}>
            <Paper
              elevation={0}
              className={[classes.paper, classes.backgroundPrimary].join(" ")}
              style={{
                paddingTop: 10,
                paddingBottom: 10
              }}
            >
              <div>
                <br />
                <Typography
                  style={{
                    color: "white"
                  }}
                  variant="h4"
                  gutterBottom
                >
                  Welcome to the best open source plataform for lean management
                </Typography>
                <Typography
                  style={{
                    color: "white"
                  }}
                  variant="h5"
                  gutterBottom
                >
                  Align strategy and execution with <b>OKRs</b> and{" "}
                  <b>Kanban</b> for you, your team and your company.
                </Typography>
              </div>
            </Paper>
          </Grid>
          <Grid item xs={12} sm={12} md={5} lg={4} xl={4}>
            <Paper
              elevation={0}
              className={classes.paper}
              style={{
                paddingTop: 10,
                paddingBottom: 10
              }}
            >
              <div style={{ margin: "0 auto" }}>
                <Typography
                  className={classes.textSuccessfull}
                  align="center"
                  variant="h4"
                  gutterBottom
                >
                  Your account has been successfully created
                </Typography>
                <br />
                <Typography align="center">
                  Click at link in the bellow to do login
                </Typography>
                <br />
                <Typography variant="h5" align="center" gutterBottom>
                  <Link className={classes.link} to="/login">
                    Login
                  </Link>
                </Typography>
              </div>
            </Paper>
          </Grid>
        </Grid>
      </Grid>
    </Grid>
  );
};

export default withStyles(styles)(AccountCreated);
