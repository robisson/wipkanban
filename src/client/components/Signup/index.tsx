// @flow
import * as React from "react";
import { withStyles, StyleRulesCallback, Theme } from "@material-ui/core/styles";
import Grid from "@material-ui/core/Grid";
import Paper from "@material-ui/core/Paper";
import Typography from "@material-ui/core/Typography";
import InputAdornment from "@material-ui/core/InputAdornment";
import TextField from "@material-ui/core/TextField";
import LockIcon from "@material-ui/icons/Https";
import EmailIcon from "@material-ui/icons/Email";
import Check from "@material-ui/icons/Check";
import Button from "@material-ui/core/Button";
import { Link, Redirect } from "react-router-dom";
import CircularProgress from "@material-ui/core/CircularProgress";
import classNames from "classnames";
import green from "@material-ui/core/colors/green";

const styles: StyleRulesCallback = (theme: Theme) => ({
  root: {
    flexGrow: 1
  },
  paper: {
    padding: theme.spacing.unit * 2,
    height: "100vh",
    display: "flex",
    alignItems: "center"
  },
  backgroundGreen: {
    backgroundColor: theme.palette.primary.main
  },
  text: {
    width: "100%"
  },
  link: {
    textDecoration: "none",
    color: theme.palette.secondary.main,
    "&:hover": {
      textDecoration: "underline"
    }
  },
  learnMore: {
    color: "white",
    borderColor: "white"
  },
  errorClass: {
    color: theme.palette.error.main
  },
  wrapper: {
    margin: theme.spacing.unit,
    position: "relative"
  },
  buttonProgress: {
    color: green[500],
    position: "absolute",
    top: "50%",
    right: 133.69 / 2 - 33 / 2,
    marginTop: -12,
    marginLeft: -12
  },
  buttonSuccess: {
    backgroundColor: green[500],
    "&:hover": {
      backgroundColor: green[700]
    }
  }
});

interface IStateSignupProps {
  classes: any;
  onCreateAccount: Function;
  success: boolean;
  showPreloader: boolean;
  message: string | null;
}

interface IStateSignup {
  email: string | null;
  password: string;
  confirmPassword: string;
  confirmPasswordError: boolean;
  requiredFields: boolean;
}

class Signup extends React.Component<IStateSignupProps, IStateSignup> {
  state: IStateSignup = {
    email: null,
    password: "",
    confirmPassword: "",
    confirmPasswordError: false,
    requiredFields: false
  };

  constructor(props: any) {
    super(props);

    this._onCreateAccount = this._onCreateAccount.bind(this);
  }

  _onCreateAccount() {
    let { password, confirmPassword } = this.state;

    if (password.length == 0 || confirmPassword.length == 0) {
      this.setState({ requiredFields: true });

      return;
    } else if (password != confirmPassword) {
      this.setState({ confirmPasswordError: true });

      return;
    }

    this.setState({ confirmPasswordError: false, requiredFields: false });
    let { onCreateAccount } = this.props;

    onCreateAccount(this.state.email, this.state.password);
  }

  render() {
    let { success, classes, message, showPreloader } = this.props;
    const buttonClassname = classNames({
      [classes.buttonSuccess]: success
    });

    if (success) {
      return <Redirect to="/account-created" />;
    }

    return (
      <Grid container className={classes.root}>
        <Grid item xs={12} className={classes.demo}>
          <Grid
            container
            spacing={0}
            className={classes.demo}
            direction="row"
            justify="flex-start"
          >
            <Grid item xs={12} sm={12} md={7} lg={8} xl={8}>
              <Paper
                elevation={0}
                className={[classes.paper, classes.backgroundGreen].join(" ")}
                style={{
                  paddingTop: 10,
                  paddingBottom: 10
                }}
              >
                <div>
                  <Typography
                    style={{
                      color: "white"
                    }}
                    variant="display1"
                    gutterBottom
                  >
                    Welcome to the best open source plataform for lean
                    management
                  </Typography>
                  <Typography
                    style={{
                      color: "white"
                    }}
                    variant="headline"
                    gutterBottom
                  >
                    Align strategy and execution with <b>OKRs</b> and{" "}
                    <b>Kanban</b> for you, your team and your company.
                  </Typography>
                </div>
              </Paper>
            </Grid>
            <Grid item xs={12} sm={12} md={5} lg={4} xl={4}>
              <Paper
                elevation={0}
                className={classes.paper}
                style={{
                  paddingTop: 10,
                  paddingBottom: 10
                }}
              >
                <div style={{ width: "80%", margin: "0 auto" }}>
                  <Typography
                    variant="display1"
                    color="primary"
                    gutterBottom
                    align="center"
                  >
                    <b>WIPLean</b>
                  </Typography>
                  <Typography component="p" gutterBottom>
                    Create your account and starts to improvement your results.
                  </Typography>
                  <div
                    style={{
                      width: "100%",
                      textAlign: "center"
                    }}
                  >
                    <TextField
                      disabled={showPreloader}
                      error={this.state.requiredFields}
                      onChange={e => {
                        this.setState({ email: e.target.value });
                      }}
                      className={classes.text}
                      label="Type your email"
                      InputProps={{
                        endAdornment: (
                          <InputAdornment position="end">
                            <EmailIcon />
                          </InputAdornment>
                        )
                      }}
                    />
                    <TextField
                      disabled={showPreloader}
                      error={
                        this.state.confirmPasswordError ||
                        this.state.requiredFields
                      }
                      onChange={e => {
                        this.setState({ password: e.target.value });
                      }}
                      type="password"
                      className={classes.text}
                      label="Password"
                      InputProps={{
                        endAdornment: (
                          <InputAdornment position="end">
                            <LockIcon />
                          </InputAdornment>
                        )
                      }}
                    />
                    <TextField
                      disabled={showPreloader}
                      error={
                        this.state.confirmPasswordError ||
                        this.state.requiredFields
                      }
                      helperText={
                        this.state.confirmPasswordError
                          ? "The password and confirm password fiedls are differents"
                          : ""
                      }
                      onChange={e => {
                        this.setState({ confirmPassword: e.target.value });
                      }}
                      type="password"
                      className={classes.text}
                      label="Confirm password"
                      InputProps={{
                        endAdornment: (
                          <InputAdornment position="end">
                            <LockIcon />
                          </InputAdornment>
                        )
                      }}
                    />
                    {message && (
                      <Typography
                        className={classes.errorClass}
                        variant="h6"
                        gutterBottom
                      >
                        {message}
                      </Typography>
                    )}
                    <div
                      style={{
                        textAlign: "right"
                      }}
                    >
                      <br />
                      <div className={classes.wrapper}>
                        <Button
                          onClick={this._onCreateAccount}
                          size="large"
                          variant="raised"
                          color="primary"
                          className={buttonClassname}
                          disabled={showPreloader}
                        >
                          Sign Up
                          <Check />
                        </Button>
                        {showPreloader && (
                          <CircularProgress
                            size={24}
                            className={classes.buttonProgress}
                          />
                        )}
                        <br />
                      </div>

                      <br />
                      <Typography
                        component="p"
                        color="primary"
                        gutterBottom
                        align="right"
                      >
                        Already have an account ?{" "}
                        <Link to="/login" className={classes.link}>
                          <b>Sign in!</b>
                        </Link>
                      </Typography>
                    </div>
                  </div>
                </div>
              </Paper>
            </Grid>
          </Grid>
        </Grid>
      </Grid>
    );
  }
}

export default withStyles(styles)(Signup);
