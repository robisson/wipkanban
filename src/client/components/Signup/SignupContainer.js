import { connect } from "react-redux";
import { createAccount } from "../../actions/User";
import Signup from "./index";

const mapDispatchToProps = dispatch => {
  return {
    onCreateAccount(email, password) {
      dispatch(createAccount(email, password));
    }
  };
};

const mapStateToProps = ({
  userReducer: { success, message, showPreloader }
}) => {
  return {
    success,
    message,
    showPreloader
  };
};

const SignupContainer = connect(
  mapStateToProps,
  mapDispatchToProps
)(Signup);

export default SignupContainer;
