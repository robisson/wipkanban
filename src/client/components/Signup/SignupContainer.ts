import { connect } from "react-redux";
import { createAccount } from "../../actions/User";
import Signup from "./index";
import { Dispatch } from "redux";

const mapDispatchToProps = (dispatch: Dispatch) => {
  return {
    onCreateAccount(email: string, password: string) {
      dispatch(createAccount(email, password));
    }
  };
};

const mapStateToProps = ({
  userReducer: { success, message, showPreloader }
}: any) => {
  return {
    success,
    message,
    showPreloader
  };
};

const SignupContainer = connect(
  mapStateToProps,
  mapDispatchToProps
)(Signup);

export default SignupContainer;
