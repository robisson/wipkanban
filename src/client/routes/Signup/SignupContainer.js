import { connect } from "react-redux";
import { createAccount } from "../../actions/User";
import Signup from "./index";

const mapDispatchToProps = dispatch => {
  return {
    onCreateAccount(email, password) {
      dispatch(createAccount(email, password));
    }
  };
};

const mapStateToProps = state => {
  return { state: state.userReducer };
};

const SignupContainer = connect(mapStateToProps, mapDispatchToProps)(
  Signup
);

export default SignupContainer;
