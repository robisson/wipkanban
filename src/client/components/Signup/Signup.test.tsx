import React from "react";
import Signup from "./index";
import { createShallow } from "@material-ui/core/test-utils";
import TextField from "@material-ui/core/TextField";
import Button from "@material-ui/core/Button";
import { Redirect } from "react-router-dom";
import { ShallowWrapper } from "enzyme";

describe("<Signup /> User", () => {
  let shallow: any;
  let wrapper: ShallowWrapper;

  const testValues: any = {
    onCreateAccount: jest.fn()
  };

  beforeEach(() => {
    shallow = createShallow({ dive: true });
    wrapper = shallow(<Signup {...testValues} />);
  });

  it("should render a component correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should render 3 TextField", () => {
    expect(wrapper.find(TextField).length).toEqual(3);
  });

  it("should render 1 Button", () => {
    expect(wrapper.find(Button).length).toEqual(1);
  });

  it("should not call function when click on button and fields are empty", () => {
    wrapper.find(Button).simulate("click");
    expect(testValues.onCreateAccount.mock.calls.length).toEqual(0);
    expect(wrapper).toMatchSnapshot();
  });

  it("should not call function when fields password and confirmPassword are different", () => {
    let expectedState: any = {
      confirmPassword: "mypassword2",
      email: "email@example.com",
      password: "mypassword",
      confirmPasswordError: true,
      requiredFields: false
    };

    let email: any = wrapper.find(TextField).at(0);
    email.simulate("change", { target: { value: "email@example.com" } });

    //password field
    wrapper
      .find(TextField)
      .at(1)
      .simulate("change", { target: { value: "mypassword" } });
    //confirm password field
    wrapper
      .find(TextField)
      .at(2)
      .simulate("change", { target: { value: "mypassword2" } });
    wrapper.find(Button).simulate("click");
    expect(testValues.onCreateAccount.mock.calls.length).toEqual(0);
    expect(wrapper.state()).toMatchObject(expectedState);
    expect(
      wrapper
        .find(TextField)
        .at(2)
        .props()
    ).toHaveProperty("error", true);
    expect(wrapper).toMatchSnapshot();
  });

  it("should call function when click on button and fields are filled", () => {
    //email field
    wrapper
      .find(TextField)
      .at(0)
      .simulate("change", { target: { value: "email@example.com" } });
    //password field
    wrapper
      .find(TextField)
      .at(1)
      .simulate("change", { target: { value: "mypassword" } });
    //confirm password field
    wrapper
      .find(TextField)
      .at(2)
      .simulate("change", { target: { value: "mypassword" } });

    wrapper.find(Button).simulate("click");
    expect(testValues.onCreateAccount.mock.calls.length).toEqual(1);
    expect(wrapper).toMatchSnapshot();
  });

  it("should to change state when type in TextField", () => {
    let expectedState: any = {
      confirmPassword: "",
      email: "email@example.com",
      password: "mypassword",
      confirmPasswordError: false,
      requiredFields: false
    };

    //email field
    wrapper
      .find(TextField)
      .at(0)
      .simulate("change", { target: { value: "email@example.com" } });
    //password field
    wrapper
      .find(TextField)
      .at(1)
      .simulate("change", { target: { value: "mypassword" } });
    expect(wrapper.state()).toEqual(expectedState);

    expect(wrapper).toMatchSnapshot();
  });

  it("should return a redirect component, bacause success prop is defined", () => {
    testValues.success = true;
    wrapper = shallow(<Signup {...testValues} />);
    expect(wrapper.type()).toEqual(Redirect);
    expect((wrapper.props() as any).to).toEqual("/account-created");
    expect(wrapper).toMatchSnapshot();
  });
});
