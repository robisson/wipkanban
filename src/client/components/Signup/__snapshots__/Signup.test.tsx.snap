// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Signup /> User should call function when click on button and fields are filled 1`] = `
<WithStyles(Grid)
  className="Signup-root-1"
  container={true}
>
  <WithStyles(Grid)
    item={true}
    xs={12}
  >
    <WithStyles(Grid)
      container={true}
      direction="row"
      justify="flex-start"
      spacing={0}
    >
      <WithStyles(Grid)
        item={true}
        lg={8}
        md={7}
        sm={12}
        xl={8}
        xs={12}
      >
        <WithStyles(Paper)
          className="Signup-paper-2 Signup-backgroundGreen-3"
          elevation={0}
          style={
            Object {
              "paddingBottom": 10,
              "paddingTop": 10,
            }
          }
        >
          <div>
            <WithStyles(Typography)
              gutterBottom={true}
              style={
                Object {
                  "color": "white",
                }
              }
              variant="display1"
            >
              Welcome to the best open source plataform for lean management
            </WithStyles(Typography)>
            <WithStyles(Typography)
              gutterBottom={true}
              style={
                Object {
                  "color": "white",
                }
              }
              variant="headline"
            >
              Align strategy and execution with
              <b>
                OKRs
              </b>
               and

              <b>
                Kanban
              </b>
               for you, your team and your company.
            </WithStyles(Typography)>
          </div>
        </WithStyles(Paper)>
      </WithStyles(Grid)>
      <WithStyles(Grid)
        item={true}
        lg={4}
        md={5}
        sm={12}
        xl={4}
        xs={12}
      >
        <WithStyles(Paper)
          className="Signup-paper-2"
          elevation={0}
          style={
            Object {
              "paddingBottom": 10,
              "paddingTop": 10,
            }
          }
        >
          <div
            style={
              Object {
                "margin": "0 auto",
                "width": "80%",
              }
            }
          >
            <WithStyles(Typography)
              align="center"
              color="primary"
              gutterBottom={true}
              variant="display1"
            >
              <b>
                WIPLean
              </b>
            </WithStyles(Typography)>
            <WithStyles(Typography)
              component="p"
              gutterBottom={true}
            >
              Create your account and starts to improvement your results.
            </WithStyles(Typography)>
            <div
              style={
                Object {
                  "textAlign": "center",
                  "width": "100%",
                }
              }
            >
              <TextField
                InputProps={
                  Object {
                    "endAdornment": <WithStyles(WithFormControlContext(InputAdornment))
                      position="end"
                    >
                      <pure(EmailIcon) />
                    </WithStyles(WithFormControlContext(InputAdornment))>,
                  }
                }
                className="Signup-text-4"
                error={false}
                label="Type your email"
                onChange={[Function]}
                required={false}
                select={false}
                variant="standard"
              />
              <TextField
                InputProps={
                  Object {
                    "endAdornment": <WithStyles(WithFormControlContext(InputAdornment))
                      position="end"
                    >
                      <pure(HttpsIcon) />
                    </WithStyles(WithFormControlContext(InputAdornment))>,
                  }
                }
                className="Signup-text-4"
                error={false}
                label="Password"
                onChange={[Function]}
                required={false}
                select={false}
                type="password"
                variant="standard"
              />
              <TextField
                InputProps={
                  Object {
                    "endAdornment": <WithStyles(WithFormControlContext(InputAdornment))
                      position="end"
                    >
                      <pure(HttpsIcon) />
                    </WithStyles(WithFormControlContext(InputAdornment))>,
                  }
                }
                className="Signup-text-4"
                error={false}
                helperText=""
                label="Confirm password"
                onChange={[Function]}
                required={false}
                select={false}
                type="password"
                variant="standard"
              />
              <div
                style={
                  Object {
                    "textAlign": "right",
                  }
                }
              >
                <br />
                <div
                  className="Signup-wrapper-8"
                >
                  <WithStyles(Button)
                    className=""
                    color="primary"
                    onClick={[Function]}
                    size="large"
                    variant="raised"
                  >
                    Sign Up
                    <pure(CheckIcon) />
                  </WithStyles(Button)>
                  <br />
                </div>
                <br />
                <WithStyles(Typography)
                  align="right"
                  color="primary"
                  component="p"
                  gutterBottom={true}
                >
                  Already have an account ?

                  <Link
                    className="Signup-link-5"
                    to="/login"
                  >
                    <b>
                      Sign in!
                    </b>
                  </Link>
                </WithStyles(Typography)>
              </div>
            </div>
          </div>
        </WithStyles(Paper)>
      </WithStyles(Grid)>
    </WithStyles(Grid)>
  </WithStyles(Grid)>
</WithStyles(Grid)>
`;

exports[`<Signup /> User should not call function when click on button and fields are empty 1`] = `
<WithStyles(Grid)
  className="Signup-root-1"
  container={true}
>
  <WithStyles(Grid)
    item={true}
    xs={12}
  >
    <WithStyles(Grid)
      container={true}
      direction="row"
      justify="flex-start"
      spacing={0}
    >
      <WithStyles(Grid)
        item={true}
        lg={8}
        md={7}
        sm={12}
        xl={8}
        xs={12}
      >
        <WithStyles(Paper)
          className="Signup-paper-2 Signup-backgroundGreen-3"
          elevation={0}
          style={
            Object {
              "paddingBottom": 10,
              "paddingTop": 10,
            }
          }
        >
          <div>
            <WithStyles(Typography)
              gutterBottom={true}
              style={
                Object {
                  "color": "white",
                }
              }
              variant="display1"
            >
              Welcome to the best open source plataform for lean management
            </WithStyles(Typography)>
            <WithStyles(Typography)
              gutterBottom={true}
              style={
                Object {
                  "color": "white",
                }
              }
              variant="headline"
            >
              Align strategy and execution with
              <b>
                OKRs
              </b>
               and

              <b>
                Kanban
              </b>
               for you, your team and your company.
            </WithStyles(Typography)>
          </div>
        </WithStyles(Paper)>
      </WithStyles(Grid)>
      <WithStyles(Grid)
        item={true}
        lg={4}
        md={5}
        sm={12}
        xl={4}
        xs={12}
      >
        <WithStyles(Paper)
          className="Signup-paper-2"
          elevation={0}
          style={
            Object {
              "paddingBottom": 10,
              "paddingTop": 10,
            }
          }
        >
          <div
            style={
              Object {
                "margin": "0 auto",
                "width": "80%",
              }
            }
          >
            <WithStyles(Typography)
              align="center"
              color="primary"
              gutterBottom={true}
              variant="display1"
            >
              <b>
                WIPLean
              </b>
            </WithStyles(Typography)>
            <WithStyles(Typography)
              component="p"
              gutterBottom={true}
            >
              Create your account and starts to improvement your results.
            </WithStyles(Typography)>
            <div
              style={
                Object {
                  "textAlign": "center",
                  "width": "100%",
                }
              }
            >
              <TextField
                InputProps={
                  Object {
                    "endAdornment": <WithStyles(WithFormControlContext(InputAdornment))
                      position="end"
                    >
                      <pure(EmailIcon) />
                    </WithStyles(WithFormControlContext(InputAdornment))>,
                  }
                }
                className="Signup-text-4"
                error={true}
                label="Type your email"
                onChange={[Function]}
                required={false}
                select={false}
                variant="standard"
              />
              <TextField
                InputProps={
                  Object {
                    "endAdornment": <WithStyles(WithFormControlContext(InputAdornment))
                      position="end"
                    >
                      <pure(HttpsIcon) />
                    </WithStyles(WithFormControlContext(InputAdornment))>,
                  }
                }
                className="Signup-text-4"
                error={true}
                label="Password"
                onChange={[Function]}
                required={false}
                select={false}
                type="password"
                variant="standard"
              />
              <TextField
                InputProps={
                  Object {
                    "endAdornment": <WithStyles(WithFormControlContext(InputAdornment))
                      position="end"
                    >
                      <pure(HttpsIcon) />
                    </WithStyles(WithFormControlContext(InputAdornment))>,
                  }
                }
                className="Signup-text-4"
                error={true}
                helperText=""
                label="Confirm password"
                onChange={[Function]}
                required={false}
                select={false}
                type="password"
                variant="standard"
              />
              <div
                style={
                  Object {
                    "textAlign": "right",
                  }
                }
              >
                <br />
                <div
                  className="Signup-wrapper-8"
                >
                  <WithStyles(Button)
                    className=""
                    color="primary"
                    onClick={[Function]}
                    size="large"
                    variant="raised"
                  >
                    Sign Up
                    <pure(CheckIcon) />
                  </WithStyles(Button)>
                  <br />
                </div>
                <br />
                <WithStyles(Typography)
                  align="right"
                  color="primary"
                  component="p"
                  gutterBottom={true}
                >
                  Already have an account ?

                  <Link
                    className="Signup-link-5"
                    to="/login"
                  >
                    <b>
                      Sign in!
                    </b>
                  </Link>
                </WithStyles(Typography)>
              </div>
            </div>
          </div>
        </WithStyles(Paper)>
      </WithStyles(Grid)>
    </WithStyles(Grid)>
  </WithStyles(Grid)>
</WithStyles(Grid)>
`;

exports[`<Signup /> User should not call function when fields password and confirmPassword are different 1`] = `
<WithStyles(Grid)
  className="Signup-root-1"
  container={true}
>
  <WithStyles(Grid)
    item={true}
    xs={12}
  >
    <WithStyles(Grid)
      container={true}
      direction="row"
      justify="flex-start"
      spacing={0}
    >
      <WithStyles(Grid)
        item={true}
        lg={8}
        md={7}
        sm={12}
        xl={8}
        xs={12}
      >
        <WithStyles(Paper)
          className="Signup-paper-2 Signup-backgroundGreen-3"
          elevation={0}
          style={
            Object {
              "paddingBottom": 10,
              "paddingTop": 10,
            }
          }
        >
          <div>
            <WithStyles(Typography)
              gutterBottom={true}
              style={
                Object {
                  "color": "white",
                }
              }
              variant="display1"
            >
              Welcome to the best open source plataform for lean management
            </WithStyles(Typography)>
            <WithStyles(Typography)
              gutterBottom={true}
              style={
                Object {
                  "color": "white",
                }
              }
              variant="headline"
            >
              Align strategy and execution with
              <b>
                OKRs
              </b>
               and

              <b>
                Kanban
              </b>
               for you, your team and your company.
            </WithStyles(Typography)>
          </div>
        </WithStyles(Paper)>
      </WithStyles(Grid)>
      <WithStyles(Grid)
        item={true}
        lg={4}
        md={5}
        sm={12}
        xl={4}
        xs={12}
      >
        <WithStyles(Paper)
          className="Signup-paper-2"
          elevation={0}
          style={
            Object {
              "paddingBottom": 10,
              "paddingTop": 10,
            }
          }
        >
          <div
            style={
              Object {
                "margin": "0 auto",
                "width": "80%",
              }
            }
          >
            <WithStyles(Typography)
              align="center"
              color="primary"
              gutterBottom={true}
              variant="display1"
            >
              <b>
                WIPLean
              </b>
            </WithStyles(Typography)>
            <WithStyles(Typography)
              component="p"
              gutterBottom={true}
            >
              Create your account and starts to improvement your results.
            </WithStyles(Typography)>
            <div
              style={
                Object {
                  "textAlign": "center",
                  "width": "100%",
                }
              }
            >
              <TextField
                InputProps={
                  Object {
                    "endAdornment": <WithStyles(WithFormControlContext(InputAdornment))
                      position="end"
                    >
                      <pure(EmailIcon) />
                    </WithStyles(WithFormControlContext(InputAdornment))>,
                  }
                }
                className="Signup-text-4"
                error={false}
                label="Type your email"
                onChange={[Function]}
                required={false}
                select={false}
                variant="standard"
              />
              <TextField
                InputProps={
                  Object {
                    "endAdornment": <WithStyles(WithFormControlContext(InputAdornment))
                      position="end"
                    >
                      <pure(HttpsIcon) />
                    </WithStyles(WithFormControlContext(InputAdornment))>,
                  }
                }
                className="Signup-text-4"
                error={true}
                label="Password"
                onChange={[Function]}
                required={false}
                select={false}
                type="password"
                variant="standard"
              />
              <TextField
                InputProps={
                  Object {
                    "endAdornment": <WithStyles(WithFormControlContext(InputAdornment))
                      position="end"
                    >
                      <pure(HttpsIcon) />
                    </WithStyles(WithFormControlContext(InputAdornment))>,
                  }
                }
                className="Signup-text-4"
                error={true}
                helperText="The password and confirm password fiedls are differents"
                label="Confirm password"
                onChange={[Function]}
                required={false}
                select={false}
                type="password"
                variant="standard"
              />
              <div
                style={
                  Object {
                    "textAlign": "right",
                  }
                }
              >
                <br />
                <div
                  className="Signup-wrapper-8"
                >
                  <WithStyles(Button)
                    className=""
                    color="primary"
                    onClick={[Function]}
                    size="large"
                    variant="raised"
                  >
                    Sign Up
                    <pure(CheckIcon) />
                  </WithStyles(Button)>
                  <br />
                </div>
                <br />
                <WithStyles(Typography)
                  align="right"
                  color="primary"
                  component="p"
                  gutterBottom={true}
                >
                  Already have an account ?

                  <Link
                    className="Signup-link-5"
                    to="/login"
                  >
                    <b>
                      Sign in!
                    </b>
                  </Link>
                </WithStyles(Typography)>
              </div>
            </div>
          </div>
        </WithStyles(Paper)>
      </WithStyles(Grid)>
    </WithStyles(Grid)>
  </WithStyles(Grid)>
</WithStyles(Grid)>
`;

exports[`<Signup /> User should render a component correctly 1`] = `undefined`;

exports[`<Signup /> User should return a redirect component, bacause success prop is defined 1`] = `
<Redirect
  to="/account-created"
/>
`;

exports[`<Signup /> User should to change state when type in TextField 1`] = `
<WithStyles(Grid)
  className="Signup-root-1"
  container={true}
>
  <WithStyles(Grid)
    item={true}
    xs={12}
  >
    <WithStyles(Grid)
      container={true}
      direction="row"
      justify="flex-start"
      spacing={0}
    >
      <WithStyles(Grid)
        item={true}
        lg={8}
        md={7}
        sm={12}
        xl={8}
        xs={12}
      >
        <WithStyles(Paper)
          className="Signup-paper-2 Signup-backgroundGreen-3"
          elevation={0}
          style={
            Object {
              "paddingBottom": 10,
              "paddingTop": 10,
            }
          }
        >
          <div>
            <WithStyles(Typography)
              gutterBottom={true}
              style={
                Object {
                  "color": "white",
                }
              }
              variant="display1"
            >
              Welcome to the best open source plataform for lean management
            </WithStyles(Typography)>
            <WithStyles(Typography)
              gutterBottom={true}
              style={
                Object {
                  "color": "white",
                }
              }
              variant="headline"
            >
              Align strategy and execution with
              <b>
                OKRs
              </b>
               and

              <b>
                Kanban
              </b>
               for you, your team and your company.
            </WithStyles(Typography)>
          </div>
        </WithStyles(Paper)>
      </WithStyles(Grid)>
      <WithStyles(Grid)
        item={true}
        lg={4}
        md={5}
        sm={12}
        xl={4}
        xs={12}
      >
        <WithStyles(Paper)
          className="Signup-paper-2"
          elevation={0}
          style={
            Object {
              "paddingBottom": 10,
              "paddingTop": 10,
            }
          }
        >
          <div
            style={
              Object {
                "margin": "0 auto",
                "width": "80%",
              }
            }
          >
            <WithStyles(Typography)
              align="center"
              color="primary"
              gutterBottom={true}
              variant="display1"
            >
              <b>
                WIPLean
              </b>
            </WithStyles(Typography)>
            <WithStyles(Typography)
              component="p"
              gutterBottom={true}
            >
              Create your account and starts to improvement your results.
            </WithStyles(Typography)>
            <div
              style={
                Object {
                  "textAlign": "center",
                  "width": "100%",
                }
              }
            >
              <TextField
                InputProps={
                  Object {
                    "endAdornment": <WithStyles(WithFormControlContext(InputAdornment))
                      position="end"
                    >
                      <pure(EmailIcon) />
                    </WithStyles(WithFormControlContext(InputAdornment))>,
                  }
                }
                className="Signup-text-4"
                error={false}
                label="Type your email"
                onChange={[Function]}
                required={false}
                select={false}
                variant="standard"
              />
              <TextField
                InputProps={
                  Object {
                    "endAdornment": <WithStyles(WithFormControlContext(InputAdornment))
                      position="end"
                    >
                      <pure(HttpsIcon) />
                    </WithStyles(WithFormControlContext(InputAdornment))>,
                  }
                }
                className="Signup-text-4"
                error={false}
                label="Password"
                onChange={[Function]}
                required={false}
                select={false}
                type="password"
                variant="standard"
              />
              <TextField
                InputProps={
                  Object {
                    "endAdornment": <WithStyles(WithFormControlContext(InputAdornment))
                      position="end"
                    >
                      <pure(HttpsIcon) />
                    </WithStyles(WithFormControlContext(InputAdornment))>,
                  }
                }
                className="Signup-text-4"
                error={false}
                helperText=""
                label="Confirm password"
                onChange={[Function]}
                required={false}
                select={false}
                type="password"
                variant="standard"
              />
              <div
                style={
                  Object {
                    "textAlign": "right",
                  }
                }
              >
                <br />
                <div
                  className="Signup-wrapper-8"
                >
                  <WithStyles(Button)
                    className=""
                    color="primary"
                    onClick={[Function]}
                    size="large"
                    variant="raised"
                  >
                    Sign Up
                    <pure(CheckIcon) />
                  </WithStyles(Button)>
                  <br />
                </div>
                <br />
                <WithStyles(Typography)
                  align="right"
                  color="primary"
                  component="p"
                  gutterBottom={true}
                >
                  Already have an account ?

                  <Link
                    className="Signup-link-5"
                    to="/login"
                  >
                    <b>
                      Sign in!
                    </b>
                  </Link>
                </WithStyles(Typography)>
              </div>
            </div>
          </div>
        </WithStyles(Paper)>
      </WithStyles(Grid)>
    </WithStyles(Grid)>
  </WithStyles(Grid)>
</WithStyles(Grid)>
`;
